src/main/java/com/flowpowered/noise/module/Module.java

```diff
@@ -26,44 +26,115 @@
+    }
 
-        // Create an array of pointers to all source modules required by this
-        // noise module.  Set these pointers to NULL.
-        if (sourceModuleCount > 0) {
-            sourceModule = new Module[sourceModuleCount];
-            for (int i = 0; i < sourceModuleCount; i++) {
-                sourceModule[i] = null;
-            }
-        } else {
-            sourceModule = null;
-        }
+    public Max max(Module that) {
+        return new Max(this, that);
     }
 
-    public Module getSourceModule(int index) {
-        if (index >= getSourceModuleCount() || index < 0 || sourceModule[index] == null) {
-            throw new NoModuleException();
-        }
-        return (sourceModule[index]);
+    public Min min(Module that) {
+        return new Min(this, that);
     }
 
-    public void setSourceModule(int index, Module sourceModule) {
-        if (this.sourceModule == null) {
-            return;
-        }
-        if (index >= getSourceModuleCount() || index < 0) {
-            throw new IllegalArgumentException("Index must be between 0 and GetSourceMoudleCount()");
-        }
-        this.sourceModule[index] = sourceModule;
+    public Multiply multiply(Module that) {
+        return new Multiply(this, that);
     }
 
-    public abstract int getSourceModuleCount();
+    public Power power(Module that) {
+        return new Power(this, that);
+    }
+
+    // Misc methods
+    public Cache cache() {
+        return new Cache(this);
+    }
+
+    // Modifier methods
+    public Abs abs() {
+        return new Abs(this);
+    }
+
+    public Clamp clamp(double lowerBound, double upperBound) {
+        return new Clamp(this, lowerBound, upperBound);
+    }
+
+    public Curve curve(Curve.ControlPoint... controlPoints) {
+        return new Curve(this, controlPoints);
+    }
+
+    public Curve curve(Collection<Curve.ControlPoint> controlPoints) {
+        return new Curve(this, controlPoints);
+    }
+
+    public Exponent exp(double exponent) {
+        return new Exponent(this, exponent);
+    }
+
+    public Invert invert() {
+        return new Invert(this);
+    }
+
+    public ScaleBias scaleBias(double bias, double scale) {
+        return new ScaleBias(this, scale, bias);
+    }
+
+    public Terrace terrace(double[] controlPoints, boolean invertTerraces) {
+        return new Terrace(this, controlPoints, invertTerraces);
+    }
+
+    // Transformer methods
+
+    // Selector methods
+
+    public Blend blendFrom(Module sourceA, Module sourceB) {
+        return new Blend(this, sourceA, sourceB);
+    }
+
+    public Select selectFrom(Module sourceA, Module sourceB, double edgeFalloff, double lowerBound, double upperBound) {
+        return new Select(this, sourceA, sourceB, edgeFalloff, lowerBound, upperBound);
+    }
+
+    public Select.Builder selectFrom(Module sourceA, Module sourceB) {
+        return new Select.Builder()
+                .setSources(sourceA, sourceB);
+    }
+
```

Comment:
The reflection method might be nice since it prevents from breaking the builder flow and I don't think it would look that bad. Not too complicated to implement either.